internal/validation/gateway/httproute.go

```diff
@@ -181,3 +187,55 @@ func getListenersForHTTPRouteValidation(sectionName *gatewayv1beta1.SectionName,
 
 	return listenersForValidation, nil
 }
+
+func validateWithKongGateway(
+	ctx context.Context, routesSvc kong.AbstractRouteService, parserFeatures parser.FeatureFlags, kongVersion semver.Version, httproute *gatewayv1beta1.HTTPRoute,
```

Comment:
I'd advocate for trying to decouple this func signature from `kong.AbstractRouteService` and hence using an `interface` which we define here.